libsolidity/inlineasm/AsmCodeGen.cpp

```diff
@@ -197,7 +185,29 @@ class CodeTransform: public boost::static_visitor<>
 	void operator()(Label const& _label)
 	{
 		m_state.assembly.setSourceLocation(_label.location);
-		m_state.assembly.append(m_state.labels.at(_label.name));
+		solAssert(m_scope.identifiers.count(_label.name), "");
+		Scope::Label& label = boost::get<Scope::Label>(m_scope.identifiers[_label.name]);
+		assignLabelIdIfUnset(label);
+		if (label.resetStackHeight)
+		{
+			size_t numVariables = boost::range::count_if(
+				 m_scope.identifiers | boost::adaptors::map_values,
+				[](Scope::Identifier const& var) { return var.type() == typeid(Scope::Variable) && boost::get<Scope::Variable>(var).active; }
+			);
+			for (auto const& identifier: _label.stackInfo)
+				if (!identifier.empty())
+				{
+					solAssert(m_scope.identifiers.count(identifier), "");
+					Scope::Variable& var = boost::get<Scope::Variable>(m_scope.identifiers[identifier]);
+					var.active = true;
+					var.stackHeight = m_initialDeposit + numVariables;
```

Comment:
Code at a different place ensures that `resetStackHeight` is set if `_label.stackInfo` is non-empty, but I can move this loop out of the if. in this case, I would first reset the stack height if requested and then register the new variables starting from the then-current stack height.